src/mo_interpreter/interpret.ml

```diff
@@ -508,10 +508,11 @@ and interpret_exp_mut env exp (k : V.value V.cont) =
       match v1 with
       | V.Obj fs ->
         k (find id.it fs)
-      | V.Blob aid when T.sub exp1.note.note_typ (T.Obj (T.Actor, [])) ->
+      | V.Blob aid as actor when T.sub exp1.note.note_typ (T.Obj (T.Actor, [])) ->
         begin match V.Env.find_opt aid !(env.actor_env) with
-        (* not quite correct: On the platform, you can invoke and get a reject *)
-        | None -> trap exp.at "Unknown actor \"%s\"" aid
+        | None ->
+          (* method not defined yet, just pair them up *)
+          k V.(Tup [actor; Text id.it])
```

Comment:
I would simplify the outer case on DotE into two cases, one on actor types, and one on other types.
The first just pairs the actor blob with the field name, the second just reads the field from the actor.
```
| DotE (exp1, id)  when T.sub exp1.note.note_typ (T.Obj (T.Actor, []))  ->
     interpret_exp env exp1 (fun v1 ->
     k (Tup [v1; Text id.it])
| DotE (exp1, id) ->
    interpret_exp env exp1 (fun v1 ->
      let fs = V.as_obj v1 in in
        k (find id.it fs)
```
Hopefully you can then just special case the application on actor fields accordingly (issuing rejects on undefined actor or reference)

Reply:
54bb508e4